src/shrinkray/state.py

```diff
@@ -152,17 +166,27 @@ def get_current_output_path(self) -> str | None:
     def get_active_test_id(self) -> int | None:
         """Get the currently running test ID, if any.
 
-        Returns the active test ID if one is running, None otherwise.
+        Returns the active test ID if one is running AND has produced output.
+        This ensures the UI doesn't flash briefly to an empty test.
         """
         if self._active_outputs:
-            return max(self._active_outputs.keys())
+            max_id = max(self._active_outputs.keys())
+            active_path = self._active_outputs[max_id]
+            if self._file_has_content(active_path):
+                return max_id
+        return None
+
+    def get_last_return_code(self) -> int | None:
+        """Get the return code of the most recently completed test."""
+        if self._completed_outputs:
+            return self._completed_outputs[-1][3]
```

Comment:
Why is this what you're showing? "Most recently completed test" isn't the right thing at all, you need to be showing the return code that corresponds to the output you're showing.